modules/security-context-constraints-psa-rectifying.adoc

```diff
@@ -0,0 +1,13 @@
+// Module included in the following assemblies:
+//
+// * authentication/managing-security-context-constraints.adoc
+
+:_content-type: PROCEDURE
+[id="security-context-constraints-psa-rectifying_{context}"]
+= About pod security admission alerts 
+
+// TODO: Anything more?
+
+A `PodSecurityViolation` alert is triggered when the Kubernetes API server reports that there is a pod denial on the audit level of the pod security admission controller.
+
+View the audit logs to investigate alerts that were triggered. As an example, a workload is likely to fail admission if global enforcement is set to the `restricted` pod security level.
```

Comment:
Please don't, the above is wrong. One should never care which SCC gets assigned to their workloads, they only ever need to make sure their security context is properly configured so that the workload is always capable of running.

If no SCC matches the workload, they:
a) need to ask admins for more SCC privileges
b) if they are the admin, they might need to create an SCC allowing such class of workloads

Besides:
a) this chapter is not about SCCs
b) customers don't have access to the cluster bot as far as I know.

Reply:
@stlaz 

`One should never care which SCC gets assigned to their workloads,`

That is not true. Operator Authors provide solutions for OCP therefore they need to test and ensure that their solutions are working well. In this way, they also need to understand how they can check what is the SCC qualified for a workload.

As an admin, if I am troobolshoting I scenario for many reasons I also need to know how I check what is the SCC that some workload is using.

Reply:
Unless this content is absolutely critical for release, it's too late to add it. Docs freeze is today. 

It can be added post-GA as a docs bug fix or enhancement, depending on whichever category best fits.

Reply:
@stlaz 

>  customers don't have access to the cluster bot as far as I know.
It was highlighted that the steps need to share the intention here is NOT to add this content exactly. Therefore, for sure we should not add a slack bot and just replace that for something like on the cluster can you check that... 

>  One should never care which SCC gets assigned to their workloads, 

I cannot go along with it. For many reasons I might want/need to know what is the SCC that has been used by a workload. So, why cannot add something like:

If you want to check what the SCC assigned to the workload then you can check for the annotation `openshift.io/scc: ` such as in the following example? Why cannot provide valid info for users to be able to troubleshoot and check these scenarios? Not that in many circumstances I might want to know what is the SCC used before for example taking some action.

Reply:
> That is not true. Operator Authors provide solutions for OCP therefore they need to test and ensure that their solutions are working well. In this way, they also need to understand how they can check what is the SCC qualified for a workload.

Actually, that is the truth.

Operator authors might need to check that their solution is running in OCP, and if it doesn't, they need to check what to do in order to make it run. They should always strive to express the security requirements of their operators and operands within their security contexts to the authors' best knowledge according to the least privilege principle.

> As an admin, if I am troobolshoting I scenario for many reasons I also need to know how I check what is the SCC that some workload is using.

You may need that in order to understand why your workload fails when an SCC mutates it so that it misses system-level permissions that it requires. This, however, only shows that your security context was not set properly, and therefore the knowledge of the SCC only helps you in a way to understand why your workload was mutated in the way it was.

Reply:
@stlaz spoke with me that would be better in another place as well. By sure I can provide a better text either. I will open a BZ. thank you @maxwelldb for the attention